sdk/storage/azure-storage-queue/azure/storage/queue/_models.py

```diff
@@ -179,74 +309,83 @@ class AccessPolicy(GenAccessPolicy):
     both in the Shared Access Signature URL and in the stored access policy, the
     request will fail with status code 400 (Bad Request).
 
-    :param str permission:
+    :param Optional[QueueSasPermissions] permission:
         The permissions associated with the shared access signature. The
         user is restricted to operations allowed by the permissions.
         Required unless an id is given referencing a stored access policy
         which contains this field. This field must be omitted if it has been
         specified in an associated stored access policy.
-    :param expiry:
+    :param Optional[Union["datetime", str]] expiry:
         The time at which the shared access signature becomes invalid.
         Required unless an id is given referencing a stored access policy
         which contains this field. This field must be omitted if it has
         been specified in an associated stored access policy. Azure will always
         convert values to UTC. If a date is passed in without timezone info, it
         is assumed to be UTC.
-    :type expiry: ~datetime.datetime or str
-    :param start:
+    :param Optional[Union["datetime", str]] start:
         The time at which the shared access signature becomes valid. If
         omitted, start time for this call is assumed to be the time when the
         storage service receives the request. Azure will always convert values
         to UTC. If a date is passed in without timezone info, it is assumed to
         be UTC.
-    :type start: ~datetime.datetime or str
     """
 
-    def __init__(self, permission=None, expiry=None, start=None):
+    permission: Optional[QueueSasPermissions] #type: ignore [assignment]
+    """The permissions associated with the shared access signature. The user is restricted to
+        operations allowed by the permissions."""
+    expiry: Optional[Union["datetime", str]] #type: ignore [assignment]
+    """The time at which the shared access signature becomes invalid."""
+    start: Optional[Union["datetime", str]] #type: ignore [assignment]
+    """The time at which the shared access signature becomes valid."""
+
+    def __init__(
+        self, permission: Optional[QueueSasPermissions] = None,
+        expiry: Optional[Union["datetime", str]] = None,
+        start: Optional[Union["datetime", str]] = None
+    ) -> None:
         self.start = start
         self.expiry = expiry
         self.permission = permission
 
 
 class QueueMessage(DictMixin):
```

Comment:
So this is where I've landed for this and I'm surprised that it's not complaining that `id = None` in some cases?

BUT this is where I landed as:

- Still allows for content to come in as a parameter
- Typehints id as non-optional `str`
- Since I needed to let kwargs through for `id`, thought we might as well add in kwargs support for the other fields so we can eventually get out of the pattern of making an empty object and setting its fields and just constructing it with the fields populated using the ctor. 😄